src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.dsde.rawls.model
+
+import org.broadinstitute.dsde.rawls.model.WorkspaceAccessLevel._
+
+sealed trait UserAuthRef
+case class RawlsUserRef(userEmail: String) extends UserAuthRef
+case class RawlsGroupRef(groupName: String) extends UserAuthRef
```

Comment:
I think it's either both or neither. Your explanation seems legit. Will need to catch fat-finger errors in email addresses though.